openpype/hosts/fusion/plugins/publish/validate_saver_resolution.py

```diff
@@ -87,19 +38,80 @@ def process(self, instance):
 
     @classmethod
     def get_invalid(cls, instance):
-        resolution = cls.get_resolution(instance)
+        saver = instance.data["tool"]
+        try:
+            resolution = cls.get_resolution(instance)
+        except PublishValidationError:
+            resolution = None
         expected_resolution = cls.get_expected_resolution(instance)
         if resolution != expected_resolution:
-            saver = instance.data["tool"]
             return [saver]
 
     @classmethod
     def get_resolution(cls, instance):
         saver = instance.data["tool"]
         first_frame = instance.data["frameStartHandle"]
-        return get_tool_resolution(saver, frame=first_frame)
+        return cls.get_tool_resolution(saver, frame=first_frame)
 
     @classmethod
     def get_expected_resolution(cls, instance):
         data = instance.data["assetEntity"]["data"]
         return data["resolutionWidth"], data["resolutionHeight"]
+
+    @classmethod
+    def get_tool_resolution(cls, tool, frame):
+        """Return the 2D input resolution to a Fusion tool
+
+        If the current tool hasn't been rendered its input resolution
+        hasn't been saved. To combat this, add an expression in
+        the comments field to read the resolution
+
+        Args
+            tool (Fusion Tool): The tool to query input resolution
+            frame (int): The frame to query the resolution on.
+
+        Returns:
+            tuple: width, height as 2-tuple of integers
+
+        """
+        if tool["Comments"][frame] is None:
+            raise PublishValidationError(
+                "Cannot get resolution info for frame '{}'.\n\n "
+                "Please check that saver has connected input.".format(
+                    frame
+                )
+            )
```

Comment:
Wait a second - this seems like a bug. Isn't this ALWAYS `None` because the code below that 'evaluates' it isn't triggered yet?

Also, you're raising a PublishValidationError to then ignore the error completely in the `get_resolution` call? Meaning this actual error **never** gets propagated.

Basically it means that WITH this PR the resolution is never validated. (Unless maybe by pure luck it's a tool where the user had himself specified a "comment" on the tool and thus it, prior to the code below does not evaluate to `None`?

Reply:
It wasn't always None, but it might be only because some leftovers from previous Publish. 

I moved check after script resolution.